### ISSUE TITLE
No response from callback url

### ISSUE THREAD

**Issue by kamandemwangi** [safaricom/mpesa-php-sdk#51]
I am interacting with lipa na M-pesa online endpoint, https://sandbox.safaricom.co.ke/mpesa/stkpush/v1/processrequest. I am experiencing an issues where sometimes I get response at my callback url but sometime there is no response at all and my callback url is accessible from the internet.

**Comment by sandeshdasi** [comment]
For me also same issue with sandbox api. sometimes getting the below response, most of the times, no response.
stkCallback: {
      MerchantRequestID: '53785-65856915-1',
      CheckoutRequestID: 'ws_CO_23052022122137653708374149',
      ResultCode: 1019,
      ResultDesc: 'Transaction has expired'
    }

**Comment by kamandemwangi** [comment]
@mossey please give as an explanation or a solution to the issue. I rely on the data posted to the callback url for systems functionality.
@sandeshdasi Besides lack of response, what you are getting means an error occurred or the transaction failed as per the Safaricom daraja documentation "ResultCode: This is a numeric status code that indicates the status of the transaction processing. 0 means successful processing and any other code means an error occurred or the transaction failed."

**Comment by sandeshdasi** [comment]
@kamandemwangi thanks for the response. I tried with LIVE details instead of SANDBOX, it is working fine, getting valid response every time.

**Comment by em-manuel** [comment]
I am simulating a C2B transaction on sandbox. The following are the steps I have taken:

1. Get auth token.
2. Register Callback URLs
3. Simulate payment.
All the above are successful.

However the callback URL is not getting any hits. It is not a UFW issue cause another STK callback script running (m-pesa live) is getting hits of the callback.

What could I be missing here?

**Comment by ronjohnson-owuor** [comment]
am also getting this issue on sandbox,Everything is happening correctly and the payment are being made successfull but my callback is not getting any hit at all from mpesa

**Comment by ronjohnson-owuor** [comment]
really urgent someone please

**Comment by qualebs** [comment]
This is an issue from safaricom's side. It has been bothering so many people for so long. It is really sad to see that 2 years later nothing has been done to fix it.

**Comment by MosesMuiru** [comment]
Hello ,
Try using alternatives of ngrok like localtunnel
works for me

**Comment by sichshiri1** [comment]
Who has been able to get the data in the callback url using asp.net? I still cannot get data to be saved in my database when live

**Comment by koechian** [comment]
> Hello , Try using alternatives of ngrok like localtunnel works for me

Had the same issue with NodeJS and switching to localtunnel sorted it out on the Sandbox API.

**Comment by JAPHETHNYARANGA** [comment]
I have the same issue , it seems the sandbox has an issue calling the callback url

**Comment by thatboywest** [comment]
> > Hello , Try using alternatives of ngrok like localtunnel works for me
> 
> Had the same issue with NodeJS and switching to localtunnel sorted it out on the Sandbox API.

am using localtunnel, it works sometimes then i just stops i think its an issue with the sandbox

**Comment by kkagunda** [comment]
Getting no response at my callback since 23rd August. stk push and payment is successful but no response from https://api.safaricom.co.ke/mpesa/stkpush/v1/processrequest. My environment hasn't changed. What could be the issue?

**Comment by kelwinsj** [comment]
It appears that this issue could still be active as I am also not getting any hits at the callbackURL. I am running some automations and hence cannot setup Ngrok in this case.

Hope this gets fixed soon-er. Otherwise, I may have to now add a query on top of my flow to check transaction status which shouldn't be necessary if this was working alright!

**Comment by kelwinsj** [comment]
Update: I was able to get the callback URL to work. I just needed to register it first as Safaricom will not send a response to a callback URL that is not registered.

You can learn more about this here: https://developer.safaricom.co.ke/SelfServices?tab=urlmanagement

**Comment by JAPHETHNYARANGA** [comment]
> Getting no response at my callback since 23rd August. stk push and payment is successful but no response from https://api.safaricom.co.ke/mpesa/stkpush/v1/processrequest. My environment hasn't changed. What could be the issue?

did you solve?

**Comment by kkagunda** [comment]
> > Getting no response at my callback since 23rd August. stk push and payment is successful but no response from https://api.safaricom.co.ke/mpesa/stkpush/v1/processrequest. My environment hasn't changed. What could be the issue?
> 
> did you solve?

Yes. The issue was Safaricom had blacklisted me (for reasons unknown to me. Probably blanket black listing after a hack attempt). Once I was white listed. The process was successful

**Comment by aiden-riddler** [comment]
M-Pesa's API rejects URLs containing certain keywords like "MPESA", verify your callback url does not contain such keywords

**Comment by pneumaticos** [comment]
Has anyone been able to get a fix for this?

**Comment by chriis-fr** [comment]
why do we keep getting 504 errors when trying to connect or login to daraja?? in production??
mediocre software with zero responses on email.

**Comment by tomleejumah** [comment]
haha well well well, nimefikiwa bana, callback was working then it just stopped im still on sandbox
